File: Bedu/Java/Clase6/ScrollView/app/src/main/java/com/charlye934/scrollview/MainActivity.java
package com.charlye934.scrollview;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    Map<String,Integer> animalesList = new HashMap<>();
    String animales[] = {"caballo","cabra","cerdo","gallo","mono","perro", "serpiente","tigre"};
    int sonidoAnimales[] = {R.raw.caballo,R.raw.cabra,R.raw.cerdo,R.raw.gallo,R.raw.mono,R.raw.perro,R.raw.serpiente,R.raw.tigre};
    MediaPlayer sonidos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        for(int i=0; i<animales.length; i++){
            animalesList.put(animales[i],sonidoAnimales[i]);
        }

        //Pondremos el icono en el action bar
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setIcon(R.mipmap.ic_launcher_foreground);
    }

    public void seleccionaAnimal(View view){
        ImageButton v = (ImageButton)view;
        String animal = String.valueOf(v.getTag());
        ejecutaSonido(animal);
    }

    public void ejecutaSonido(String nombreAnimal){
        int ani = animalesList.get(nombreAnimal);
        if(sonidos != null ){
            sonidos.stop();
        }
        sonidos = MediaPlayer.create(getApplicationContext(),ani);
        sonidos.start();
    }
}

File: Bedu/Kotlin/Clase1-6/ScrollViewAnimales/app/src/main/java/com/example/scrollviewanimales/MainActivity.kt
package com.example.scrollviewanimales

import android.media.MediaPlayer
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    var animaList:Map<String,Int> = mapOf("caballo" to R.raw.caballo,"cabra" to R.raw.cabra,"cerdo" to R.raw.cerdo,"gallo" to R.raw.gallo,"mono" to R.raw.mono,"perro" to R.raw.perro, "serpiente" to R.raw.serpiente,"tigre" to R.raw.tigre)
    var sonidos:MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Pondremos el icono en el action bar
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.setIcon(R.mipmap.ic_launcher_foreground)
    }

    fun seleccionaAnimal(view:View){
        val animal = view.tag.toString()
        ejecutaSOnido(animal)
    }

    fun ejecutaSOnido(nombreAnimal:String){
        var animal = animaList.get(nombreAnimal)
        if(sonidos != null){
            sonidos!!.stop()
        }
        sonidos = MediaPlayer.create(applicationContext,animal!!)
        sonidos!!.start()
    }


}
